File: java/src/test/java/kms/ranking/TestGetSeedRanking.java
package kms.ranking;

import dev.spiralmoon.maplestory.api.common.MapleStoryApiErrorCode;
import dev.spiralmoon.maplestory.api.common.MapleStoryApiException;
import dev.spiralmoon.maplestory.api.kms.MapleStoryApi;
import dev.spiralmoon.maplestory.api.kms.dto.ranking.TheSeedRankingResponseDTO;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import java.time.LocalDateTime;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

@DisplayName("getSeedRanking")
public class TestGetSeedRanking {
    private static final String apiKey = System.getProperty("API_KEY_KMS"); // Your API Key
    private static final MapleStoryApi api = new MapleStoryApi(apiKey);
    private static final String ocid = "e0a4f439e53c369866b55297d2f5f4eb";

    @Test
    @DisplayName("success: getSeedRanking")
    void getSeedRanking() {
        TheSeedRankingResponseDTO response = api.getTheSeedRanking(null, null, null).join();
        System.out.println(response.toString());
    }

    @Test
    @DisplayName("success: getSeedRanking with paging")
    void getSeedRanking_with_paging() {
        TheSeedRankingResponseDTO response = api.getTheSeedRanking(null, null, 2).join();
        System.out.println(response.toString());
    }

    @Test
    @DisplayName("success: getSeedRanking with ocid")
    void getSeedRanking_with_ocid() {
        TheSeedRankingResponseDTO response = api.getTheSeedRanking(null, ocid, null).join();
        assertThat(response.getRanking()).hasSizeGreaterThanOrEqualTo(0);
        System.out.println(response.toString());
    }

    @Test
    @DisplayName("success: getSeedRanking with world name")
    void getSeedRanking_with_world_name() {
        TheSeedRankingResponseDTO response = api.getTheSeedRanking("스카니아", null, null).join();
        System.out.println(response.toString());
    }

    @Test
    @DisplayName("success: getSeedRanking with date")
    void getSeedRanking_with_date() {
        LocalDateTime date = LocalDateTime.of(2023, 12, 23, 0, 0);
        TheSeedRankingResponseDTO response = api.getTheSeedRanking(null, null, null, date).join();
        System.out.println(response.toString());
    }

    @Test
    @DisplayName("fail: getSeedRanking with invalid date")
    void getSeedRanking_with_invalid_date() {
        LocalDateTime invalidDate = LocalDateTime.of(2023, 12, 21, 0, 0);
        assertThatThrownBy(() -> api.getTheSeedRanking(null, null, null, invalidDate).join())
                .hasCauseInstanceOf(IllegalArgumentException.class)
                .satisfies(e -> {
                    Throwable error = e.getCause();
                    assertThat(error.getMessage()).contains("You can only retrieve data after 2023-12-22.");
                    System.out.println(error.getMessage());
                });
    }

    @Test
    @DisplayName("fail: getSeedRanking with invalid ocid throw OPENAPI00003")
    void getSeedRanking_with_invalid_ocid_throw_OPENAPI00003() {
        final String invalidOcid = "invalid_ocid_123";
        assertThatThrownBy(() -> api.getTheSeedRanking(null, invalidOcid, null).join())
                .hasCauseInstanceOf(MapleStoryApiException.class)
                .satisfies(e -> {
                    MapleStoryApiException apiException = (MapleStoryApiException) e.getCause();
                    assertThat(apiException.getErrorCode()).isEqualTo(MapleStoryApiErrorCode.OPENAPI00003);
                    System.out.println(apiException.getErrorCode() + " " + apiException.getMessage());
                });
    }
}

File: csharp/MapleStory.Test/KMS/Ranking/TestGetSeedRanking.cs
using MapleStory.OpenAPI.Common;
using MapleStory.OpenAPI.KMS;
using NUnit.Framework;
using System.Security.Cryptography;

namespace MapleStory.Test.KMS.Ranking
{
    [TestFixture]
    public class TestGetSeedRanking
    {
        private static readonly string apiKey = EnvConfig.Get("API_KEY_KMS"); // Your API Key
        private static readonly MapleStoryAPI api = new MapleStoryAPI(apiKey);
        private static readonly string ocid = "e0a4f439e53c369866b55297d2f5f4eb";

        [Test, Description("success: GetTheSeedRanking")]
        public async Task GetTheSeedRanking()
        {
            var response = await api.GetTheSeedRanking(worldName: null, ocid: null, page: null);
            Console.WriteLine(response.ToJson());
        }

        [Test, Description("success: GetTheSeedRanking with paging")]
        public async Task GetTheSeedRanking_With_Paging()
        {
            var response = await api.GetTheSeedRanking(worldName: null, ocid: null, page: 2);
            Console.WriteLine(response.ToJson());
        }

        [Test, Description("success: GetTheSeedRanking with ocid")]
        public async Task GetTheSeedRanking_With_Ocid()
        {
            var response = await api.GetTheSeedRanking(worldName: null, ocid, page: null);
            Assert.That(response.Ranking.Count, Is.GreaterThanOrEqualTo(0));
            Console.WriteLine(response.ToJson());
        }

        [Test, Description("success: GetTheSeedRanking with world name")]
        public async Task GetTheSeedRanking_With_World_Name()
        {
            var response = await api.GetTheSeedRanking(worldName: "스카니아", ocid: null, page: null);
            Console.WriteLine(response.ToJson());
        }

        [Test, Description("success: GetTheSeedRanking with date")]
        public async Task GetTheSeedRanking_With_Date()
        {
            var date = new DateTime(2023, 12, 23, 0, 0, 0);
            var response = await api.GetTheSeedRanking(worldName: null, ocid: null, page: null, date);
            Console.WriteLine(response.ToJson());
        }

        [Test, Description("fail: GetTheSeedRanking with invalid date")]
        public void GetTheSeedRanking_With_Invalid_Date()
        {
            var invalidDate = new DateTime(2023, 12, 21, 0, 0, 0);
            var e = Assert.ThrowsAsync<ArgumentException>(async () => await api.GetTheSeedRanking(worldName: null, ocid: null, page: null, invalidDate));
            Assert.That(e.Message, Does.Contain("You can only retrieve data after 2023-12-22."));
            Console.WriteLine(e.Message);
        }

        [Test, Description("fail: GetTheSeedRanking with invalid ocid throws OPENAPI00003")]
        public void GetTheSeedRanking_With_Invalid_Ocid_Throws_OPENAPI00003()
        {
            var invalidOcid = "invalid_ocid_123";
            var e = Assert.ThrowsAsync<MapleStoryAPIException>(async () => await api.GetTheSeedRanking(worldName: null, invalidOcid, page: null));
            Assert.That(e.ErrorCode, Is.EqualTo(MapleStoryAPIErrorCode.OPENAPI00003));
            Console.WriteLine($"{e.ErrorCode} {e.Message}");
        }
    }
}
